app/code/Magento/Sales/Model/Service/OrderService.php

```diff
@@ -267,9 +267,12 @@ public function setState(
             ]
         );
 
+        /**
+        * Event argument `transport` is @deprecated. Use `transportObject` instead.
+        */
         $this->eventManager->dispatch(
             'sales_order_state_change_before',
-            ['order' => $this, 'transport' => $transport]
+            ['order' => $this, 'transport' => $transport, 'transportObject' => $transport]
```

Comment:
```suggestion
            ['order' => $this, 'transport' => $transport, 'order_object' => $order]
```
To address the concern in the reported issue the order object should be made available via the event.